optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/solver/score/BatchSchedulingEasyScoreCalculator.java

```diff
@@ -0,0 +1,244 @@
+                    hard1Score -= entry.getValue();
+                }
+            }
+        }
+
+        for (Map.Entry<Long, Long> entry : batchOtherPenaltyValueMap.entrySet()) {
+            if ((entry.getValue() > 0) && (batchRoutePathMap.get(entry.getKey()) != null)) {
+                hard0Score -= OTHER_ALLOCATION_PENALTY;
+
+                if (COMPUTE_hard1Score) {
+                    hard1Score -= entry.getValue();
+                }
+            }
+        }
+
+        for (Map.Entry<String, Long> entry : segmentOverlapMap.entrySet()) {
+            hard2Score -= entry.getValue();
+        }
+
+        for (Allocation allocation : schedule.getAllocationList()) {
+            if (batchRoutePathMap.get(allocation.getBatch().getId()) != null) {
+                if (allocation.getRoutePath().getPath().equals(batchRoutePathMap.get(allocation.getBatch().getId()))) {
+                    if ((allocation.getEndInjectionTime() != null) && (allocation.getEndInjectionTime() > soft0Score)) {
+                        soft0Score = allocation.getEndInjectionTime();
+                    }
+                }
+            }
+        }
+
+        soft0Score = -soft0Score;
+
+        Map<Long, Long> map1 = new HashMap<Long, Long>();
+
+        for (Allocation allocation : schedule.getAllocationList()) {
+
+            if (batchRoutePathMap.get(allocation.getBatch().getId()) != null) {
+                if (allocation.getRoutePath().getPath().equals(batchRoutePathMap.get(allocation.getBatch().getId()))) {
+                    if ((allocation.getEndInjectionTime() != null)) {
+                        if (map1.get(allocation.getBatch().getId()) != null) {
+                            if (allocation.getEndInjectionTime() > map1.get(allocation.getBatch().getId())) {
+                                map1.put(allocation.getBatch().getId(), allocation.getEndInjectionTime());
+                            }
+                        } else {
+                            map1.put(allocation.getBatch().getId(), allocation.getEndInjectionTime());
+                        }
+                    }
+                }
+            }
+        }
+
+        for (Map.Entry<Long, Long> entry : map1.entrySet()) {
+            soft1Score -= entry.getValue();
+        }
+
+        if (COMPUTE_soft2Score) {
+            soft2Score = -soft2Score;
+        } else {
+            soft2Score = 0;
+        }
+
+        return BendableLongScore.of(new long[] { hard0Score, hard1Score, hard2Score },
+                new long[] { soft0Score, soft1Score, soft2Score });
+    }
+
+    public String generateCompositeKey(String key1, String key2) {
+        return key1 + "#" + key2;
+    }
+
+    public String generateKey(Allocation allocation) {
+        return allocation.getBatch().getName() + ":" + allocation.getRoutePath().getPath() + ":"
+                + allocation.getSegment().getName();
+    }
+
+    public String[] getSegmentArray(String routePath) {
```

Comment:
Any chance we could reuse code from the other calculator?
If it does the same thing, it should be the same method.

Reply:
I'll refactor the code and will extract all common methods from Easy and Incremental calculator (includes calculation logic) in a common class. 

After refactoring, hopefully the calculation logic will be simplified.